### PR TITLE
improvement: status type | edit/create mode

### PR DESCRIPTION
**Pull request by CREDO23**
## Description

FIX #3184 

## Type of Change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Breaking change
-   [ ] Documentation update

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings

## Previous screenshots

[Loom](https://www.loom.com/share/8f8f756fcbef4acd9848f1374eb6e7ec?sid=cdd48f78-06bd-4267-933e-e7f4acd2f6ee)

## Current screenshots

[Loom](https://www.loom.com/share/fc0bf8ae4d6244e7ab0d01d67b9dc192?sid=b823978c-a8ff-4d03-97a1-b883866ae894)

### CODE REVIEW

**Comment by evereq** [comment]
@CREDO23 :
1. Please review suggestions by Coderabbit and resolve each one (adjust code or close each one)
2. Please look in picture below, why selector for status has different default color compared to other, e.g. icon selector!? 

image

**Comment by CREDO23** [comment]
> @CREDO23 :
> 
> 1. Please review suggestions by Coderabbit and resolve each one (adjust code or close each one)
> 2. Please look in picture below, why selector for status has different default color compared to other, e.g. icon selector!?
> 
> image

@evereq , @KostyaEver suggested stopping working on this ! I will continue after having a call with him and @GloireMutaliko21 !
So I will keep this as a draft for now !

**Review by Cedric921 — CHANGES_REQUESTED** [review]
I've just reviewed it, 
The RabbitAI refactor suggestions is very pertinent (especially the ones I've stuck with 👍🏽)  , When you will continue to work on this Task, try to integrate/fix them. 
@CREDO23

File: apps/web/lib/settings/task-statuses-form.tsx [hunk 1]
```diff
@@ -153,6 +157,53 @@ export const TaskStatusesForm = ({
...
+   * @returns {string} - Random color
+   */
+  const getRandomColor = useCallback(() => {
+    const letters = '0123456789ABCDEF';
+    let color = '#';
+    for (let i = 0; i < 6; i++) {
+      color += letters[Math.floor(Math.random() * 16)];
+    }
+    return color;
+  }, []);
```
[the remark this answers is not in the document]
Comment (CREDO23) [reply apps/web/lib/settings/task-statuses-form.tsx:199]: We should have more than 5 colors, so I think, for now, we can keep generating them randomly

**Comment by evereq** [comment]
@CREDO23 can you summarize yourself what changes this PR introduce from UX / UI point of view? How it worked before vs how it works now? Maybe record Loom or at min send screenshots please

**Comment by CREDO23** [comment]
> @CREDO23 can you summarize yourself what changes this PR introduce from UX / UI point of view? How it worked before vs how it works now? Maybe record Loom or at min send screenshots please

- Introduce the `getTextColor` function to set a suitable color regarding the background of the status item
- Fix background and size of dropdowns (status type, size, priority)
- Autos elect the icon when the user selects/changes the status type
- Auto select a random color when the user selects/changes the status type
- Add in memory caching mechanism to improve UX/performance

**Comment by evereq** [comment]
@CREDO23 what is with this PR, can it be merged / ready for review? CC: @Cedric921

**Comment by CREDO23** [comment]
> @CREDO23 what is with this PR, can it be merged / ready for review? CC: @Cedric921

Yes @evereq, it can be merged !
@Cedric921  can review it before as well !
